File: go-code/conditional_logic/replace_conditional_with_polymorphism/example1/after/NorwegianBlueParrot.go
package after

type NorwegianBlueParrot struct {
	BirdAttributes
}

func (n NorwegianBlueParrot) GetBirdAttributes() BirdAttributes {
	return n.BirdAttributes
}

func (n NorwegianBlueParrot) Plumage() string {
	if n.Voltage > 100 {
		return "scorched"
	}
	return "beautiful"
}

func (n NorwegianBlueParrot) AirSpeedVelocity() int {
	if n.IsNailed {
		return 0
	}
	return 10 + n.Voltage/10
}

File: java-code/src/main/java/refactoring/dealing_with_inheritance/replace_subclass_with_delegate/replacing_hierarchy/after/NorwegianBlueParrot.java
//10. We dont need the specific bird types anymore

/*
package refactoring.dealing_with_inheritance.replace_subclass_with_delegate.replacing_hierarchy.after;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
class NorwegianBlueParrot extends Bird {
  private int voltage;
  private boolean isNailed;

  NorwegianBlueParrot(BirdData data) {
    super(data);
    this.voltage = data.getVoltage();
    this.isNailed = data.isNailed();
  }

  @Override
  public String plumage() {
    if (voltage > 100) {
      return "scorched";
    } else {
      return super.plumage();
    }
  }

  @Override
  int airSpeedVelocity() {
    return isNailed ? 0 : 10 + voltage / 10;
  }
}
*/
